File: Data Structures I/Day 10/Binary-Tree-Postorder-Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */

const postorderTraversal = (root) => {
  const result = [];

  function postorder(root) {
    if (root === null) {
      return;
    }

    postorder(root.left);
    postorder(root.right);
    result.push(root.val);
  }

  postorder(root);

  return result;
};

File: Rust/Binary-Tree-Postorder-Traversal.rs
use std::rc::Rc;
use std::cell::RefCell;
impl Solution {
    pub fn postorder_traversal(root: Option<Rc<RefCell<TreeNode>>>) -> Vec<i32> {
        fn postorder(node: &Option<Rc<RefCell<TreeNode>>>, result: &mut Vec<i32>) {
            if let Some(v) = node {
                let v = v.borrow();               
                postorder(&v.left, result);    
                postorder(&v.right, result);
                result.push(v.val);
            }
        }

        let mut result = vec![];
        if let Some(v) = root {
            postorder(&Some(v), &mut result);
        }
        result  
    }
}